src/librustc_resolve/lib.rs

```diff
@@ -852,6 +828,24 @@ impl Module {
             pub_glob_count: Cell::new(0),
             resolved_import_count: Cell::new(0),
             populated: Cell::new(!external),
+        })
+    }
+
+    fn def_id(&self) -> Option<DefId> {
+        self.def.get().as_ref().map(Def::def_id)
+    }
+
+    fn is_normal(&self) -> bool {
```

Comment:
`Module` probably should be named something like `Scope` since it represents scopes corresponding to traits, enums, and blocks as well as actual `mod` modules.
In this case a "normal" `Module` (formerly `NormalModuleKind`) is a `Module` corresponding to a actual module or a foreign module.